pom.xml

```diff
@@ -2614,6 +2614,18 @@
       </activation>
     </profile>
 
+    <profile>
+      <id>java17</id>
+      <properties>
+        <argLine>-Xmx2g --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED --add-opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/sun.nio.cs=ALL-UNNAMED --add-opens=java.base/sun.security.action=ALL-UNNAMED --add-opens=java.base/sun.util.calendar=ALL-UNNAMED --add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED -Djol.magicFieldOffset=true</argLine>
```

Comment:
Are these args needed for running production jobs on Java 17?

Reply:
That's a good question. I think these would be needed in runtime before we can confirm Hudi doesn't use some public Java 8 APIs that later got converted to private. But I'm not sure how we can confirm that without compiling Hudi with Java 17. Maybe we can try removing some of them to see if tests fail?

Reply:
Let's keep these args for tests in the PR.  We should follow up on answer the question.